### ISSUE TITLE
Calendar Overflow discussion

### ISSUE THREAD

**Issue by aethelwulffe** [LibreHealthIO/LibreEHR#774]
There are many format errors apparent in the Calendar when viewing a large organization.

I suggest we standardize testing suite DB's to have a MINIMUM of 50 facilities and 25 providers when testing calendar or flowboard items.

**Comment by aethelwulffe** [comment]
Recent chat questions on this topic concern standard testing database resources.
1.  In EHR_contribs repo https://github.com/LibreHealthIO/EHR_Contribs/tree/master/test_sql is a largish database for testing calendar and other items.
2.  Unfortunately, we also have https://github.com/LibreHealthIO/EHR-Test-Tools/tree/master/test-data  which is a small data set.  The test_sql in contribs probably needs to get moved to test tools, but that is where resources are right now.
3.  In the forum, in the EHR section you may also find links to download the test data generator win32 application.

**Comment by aethelwulffe** [comment]
Yes, it generates facilities.
So, x12 partners are the clearinghouses that handle electronic billing and eligibility checks between the doctor's office and the insurance company.  The table that stores these is x12_partners.  Go ahead and add this to your export.  You should also add a number (probably a large number) of facilities.  I will add that feature to printapatient now and publish again.
In the meantime, here are some facilities:```
INSERT INTO `facility` (`id`, `name`, `street`, `city`, `state`, `postal_code`, `country_code`, `federal_ein`, `service_location`, `billing_location`, `accepts_assignment`, `pos_code`, `attn`, `facility_npi`, `tax_id_type`, `color`, `primary_business_entity`) VALUES,
('1','TESTCLINIC MRI','10475 ANCIENT PKWY',' N UNIT 102A','Cayo Loco','FL','922565004','USA','','1','0','1','11','Billing','9629025820','EI','#FFCC99','0'),
('2','TESTCLINIC CBO','6500 ROLLING ROAD UNIT 103','CAYO LOCO','FL','922168070','USA','123456789','1','0','1','11','Billing','9629025820','EI','#FFCC99','0'),
('3','TESTCLINIC CENTER ONE LOCATION','10475 ANCIENT PKWY N UNIT 220','CAYO LOCO','FL','922565004','USA','','1','0','1','11','Billing','9295956949','EI','#FFCC99','0'),
('4','TESTCLINIC BEACHES OFFICE','292 PURE FUN PARK DRIVE ','PURE FUN','FL','920826600','USA','','1','0','1','11','Billing','9548689909','EI','#FFCC99','0'),
('5','TESTCLINIC SHORELINE OFFICE','2627 SHORELINE AVENUE THIRD FLOOR','CAYO LOCO','FL','922044712','USA','','1','0','1','11','Billing','9629025820','EI','#FFCC99','0'),
('6','PHYSICAL THERAPY SHORELINE','2627 SHORELINE AVE STE 300','CAYO LOCO','FL','922044712','USA','','1','0','1','11','Billing','9629025820','EI','#FFCC99','0'),
('7','TESTCLINIC PURPLE PARK','2300 PARK AVENUE UNIT 203','PURPLE PARK','FL','920735573','USA','','1','0','1','11','Billing','9730500489','EI','#FFCC99','0'),
('8','PHYSICAL THERAPY C ONE','10475 ANCIENT PARKWAY NORTH STE 305','CAYO LOCO','FL','922565003','USA','','1','0','1','11','Billing','9629025820','EI','#FFCC99','0'),
('9','PHYSICAL THERAPY PURPLE PARK','2300 PARK AVENUE STE 102','PURPLE PARK','FL','920735573','USA','','1','0','1','11','Billing','9629025820','EI','#FFCC99','0'),
('10','TESTCLINIC CLAY COUNTY OFFICE','1658 ST TAFFY WAY ','MARGARITAVILLE','FL','920688447','USA','','1','0','1','11','Billing','9730500497','EI','#FFCC99','0'),
('11','PHYSICAL THERAPY BEACHES','292 PURE FUN PARK DRIVE ','PURE FUN','FL','920826600','USA','','1','0','1','11','Billing','9629025820','EI','#FFCC99','0'),
('12','ST TAFFYS WAY','1658 ST TAFFY\'S WAY ','MARGARITAVILLE','FL','920688447','USA','','1','0','1','21','Billing','9457690479','EI','#FFCC99','0'),
('13','TESTCLINIC CBO','6500 ROLLING ROAD UNIT 103','CAYO LOCO','FL','922168070','USA','123456789','1','0','1','21','Billing','9629025820','EI','#FFCC99','0'),
('14','TESTCLINIC NORTHSIDE','15255 MAX LEGGETT PKWY UNIT 5300 ','CAYO LOCO','FL','922187274','USA','','1','0','1','11','Billing','9994383023','EI','#FFCC99','0'),
('15','PARKSIDE SURGERY CENTER','2731 PARK STREET ','CAYO LOCO','FL','922047607','USA','','1','0','1','24','Billing','9295990627','EI','#FFCC99','0'),
('16','TESTCLINIC CBO','6500 ROLLING ROAD UNIT 103','CAYO LOCO','FL','922168070','USA','123456789','1','0','1','24','Billing','9629025820','EI','#FFCC99','0'),
('17','CENTER ONE SURGERY CENTER','10475 ANCIENT PKWY NORTH STE101','CAYO LOCO','FL','922565003','USA','','1','0','1','24','Billing','9609939229','EI','#FFCC99','0'),
('18','TESTCLINIC CBO','6500 ROLLING ROAD UNIT 103','CAYO LOCO','FL','922168070','USA','123456789','1','0','1','12','Billing','9629025820','EI','#FFCC99','0'),
('19','ST TAFFYS SOUTHSIDE','4201 BELFORT ROAD ','CAYO LOCO','FL','922161431','USA','','1','0','1','22','Billing','9396926747','EI','#FFCC99','0'),
('20','TESTCLINIC CBO','6500 ROLLING ROAD UNIT 103','CAYO LOCO','FL','922168070','USA','123456789','1','0','1','22','Billing','9629025820','EI','#FFCC99','0'),
('21','SURGERY CENTER OF PURPLE PARK','1465 KINGSLEY AVE UNIT 1201','PURPLE PARK','FL','920734504','USA','','1','0','1','24','Billing','9829342858','EI','#FFCC99','0'),
('22','ST TAFFYS WAY','1658 ST TAFFY\'S WAY ','MARGARITAVILLE','FL','920688447','USA','','1','0','1','22','Billing','9457690479','EI','#FFCC99','0'),
('23','ST TAFFYS SHORELINE','1 BURRCLIFF WAY ','CAYO LOCO','FL','922044748','USA','','1','0','1','21','Billing','9934997575','EI','#FFCC99','0'),
('24','ST TAFFYS SOUTHSIDE','4201 BELFORT ROAD ','CAYO LOCO','FL','922161431','USA','','1','0','1','21','Billing','9396926747','EI','#FFCC99','0'),
('25','ST TAFFYS SHORELINE','1 BURRCLIFF WAY ','CAYO LOCO','FL','922044748','USA','','1','0','1','22','Billing','9934997575','EI','#FFCC99','0'),
('26','TESTCLINIC CBO','6500 ROLLING ROAD UNIT 103','CAYO LOCO','FL','922168070','USA','123456789','1','0','1','12','Billing','9295956949','EI','#FFCC99','0'),
('27','TESTCLINIC CLAY COUNTY OFFICE','1658 ST TAFFY WAY ','MARGARITAVILLE','FL','920688447','USA','','1','0','1','22','Billing','9730500497','EI','#FFCC99','0');
```

**Comment by apooravc** [comment]
@aethelwulffe Alright. What should we do with already present data sets in contribs repo and test tool repo? I also tested [Provider Scroll](https://github.com/LibreHealthIO/lh-ehr/pull/1205) with 100 providers, the display was fine in agenda views.

**Comment by aethelwulffe** [comment]
I think I would like to do a major revision of the test data generator 
today and tomorrow to bring it in line with the output of its 
descendants, the x12 data importer tool (not public, though available to 
you, not real useful to you without input files) which has broader 
content, breaks the output up by table with VALUES() statements, and is 
(or has an option for) compatible with the database import/demo/save 
tools in the QA Measures toolset.

-Then we build a good static demo database to use.

**Comment by apooravc** [comment]
@aethelwulffe Any updates with PrintaPatient?

**Comment by aethelwulffe** [comment]
Yes, and not posted.  Well, there was one fix, but I need to get that out today.  Still working on my mail list.
  Please post a line from the bad output here to ensure I get the algorithm corrected.
